src/foam/u2/dialog/Popup.js

```diff
@@ -0,0 +1,98 @@
+/**
+ * @license
+ * Copyright 2014 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+foam.CLASS({
```

Comment:
Needs docs to explain what it does/when to use, and how it differs from PopupView.  Also, is there no way for this to be combined with PopupView?  In FOAM1 the PopupView had two uses, both supported by the same class.  I don't want to get into the same situation as FOAM1 where we had multiple similar classes for almost identical things and no clear documentation on why/when to use each.